virtual_ecosystem/core/core_components.py

```diff
@@ -188,138 +193,155 @@ class LayerStructure:
         10, "canopy", "Height of the last canopy layer ", "``PlantsModel``", "--"
         11, "surface", "Near surface conditions", ``surface_layer_height``, "0.1 m"
         12, "topsoil", "Top soil layer depth",  ``soil_layers``, "-0.25 m"
-        13, "subsoil", "First subsoil layer depth",  ``soil_layers``, "-1.25 m"
+        13, "subsoil", "First subsoil layer depth",  ``soil_layers``, "-1.00 m"
 
     .. role:: python(code)
         :language: python
 
-    The instance also provides the ``role_indices`` and ``role_indices_bool`` attributes
-    that provide dictionaries of array indices for the locations of each of the four
-    vertical layer roles. For example, given the example table above,
-    :python:`layer_structure.role_indices["topsoil"]` would return
-    :python:`np.array([12])` and :python:`layer_structure.role_indices_bool["topsoil"]`
-    would return :python:`np.array([False, ..., False, True, False])`.
+    .. py:current_module:: ~virtual_ecosystem.core.core_components
+
+    **Additional Roles**:
+        The following additional roles and attributes are also defined when the instance
+        is created.
+
+        1. The ``active_soil`` role indicates soil layers that fall even partially above
+           the configured `max_depth_of_microbial_activity`. The `soil_layer_thickness`
+           attribute provides the thickness of each soil layer - including both top- and
+           sub-soil layers - and the `soil_layer_active_thickness` records the depth of
+           biologically active soil within each layer.
+
+        2. The ``all_soil`` role is the combination of the ``topsoil`` and ``subsoil``
+           layers.
+
+        3. The ``atmosphere`` role is the combination of ``above``, ``canopy`` and
+           ``surface`` layers.
+
+        4. The ``filled_canopy`` role indicates canopy layers that contain any canopy
+           across all of the grid cells. Canopy layers below these layers do not contain
+           any canopy, and this is initialised to show no filled canopy layers. The
+           ``lowest_canopy_filled`` attribute contains an array showing the lowest
+           filled canopy layer in each grid cell. It contains ``np.nan`` when there is
+           no canopy in a grid cell and is initalised as an array of ``np.nan`` values.
+
+           Both of these are updated by the :meth:`~LayerStructure.set_filled_canopy`
+           method, which is used to dynamically update these layer components during a
+           simulation.
```

Comment:
At what point in the simulation is this updated? Will be relevant for plants/atmosphere

Reply:
That's to be determined, but I think I'd drop it into ``plants`` in the ``__init__`` to set the canopy heights from the initial plant community data  and then just re-run it in update as they grow. It won't be any different from the current flow.